File: src/main/java/com/example/Dilim/entity/NewsTr.java
package com.example.Dilim.entity;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "news_tr")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@EqualsAndHashCode(of = {"id"})
@ToString
@Builder
public class NewsTr implements Serializable {
    @Id
    @SequenceGenerator(name="seq_new_id", allocationSize = 1)
    @GeneratedValue(generator = "seq_new_id", strategy = GenerationType.SEQUENCE)
    private Long id;
    private String title;
    private String source;
    @Column(length = 500, name="description")
    private String description;
    private String createdDate;
}

File: src/main/java/com/example/Dilim/entity/NewsTr.kt
package com.example.Dilim.entity

import com.example.Dilim.dto.NewsTrDto
import lombok.AllArgsConstructor
import lombok.NoArgsConstructor
import org.hibernate.annotations.GenericGenerator
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "news_tr")
@AllArgsConstructor
@NoArgsConstructor
internal data class NewsTr(
        @Id
        @GeneratedValue(generator = "UUID")
        @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
        val id: Long?,
        val title: String ,
        val source: String,
        @Column(length = 500, name = "description")
        val description: String?,
        val createdDate: String
) {

        companion object {
                fun fromDto(dto: NewsTrDto, newsTr: NewsTr) = NewsTr (
                        id = dto.id!!,
                        title = dto.title ?: newsTr.title,
                        source = dto.source ?: newsTr.source,
                        description = dto.description ?: newsTr.source,
                        createdDate = dto.createdDate ?: newsTr.getDate()
                )
        }

        fun getDate() : String {
                val current = LocalDateTime.now()
                val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
               return current.format(formatter)
        }

}
